test/renderer_test.rb

```diff
@@ -123,6 +123,36 @@ class RendererTest < NicePartials::Test
     end
   end
 
+  test "passing local_assigns as content seeds with positional render call" do
+    byline = "Some guy"
+
+    render "local_assigns", title: "Title", byline: byline, header: tag.h1("Yo") do |partial|
+      partial.byline ", who writes stuff"
+      partial.header " more"
+    end
+
+    assert_css "h1", text: "Title"
+    assert_css "span", text: "Some guy, who writes stuff"
+    assert_match "<h1>Yo</h1> more", rendered
+
+    assert_equal "Some guy", byline # We can't mutate the passed in content.
+  end
+
+  test "passing local_assigns as content seeds with hash render call" do
+    byline = "Some guy"
+
+    # Action View won't accept passing a block to this render call style, and blankly overrides the passed in `:partial` key
+    # …for some reason.
+    # https://github.com/rails/rails/blob/90a272a1de6f13711943139c4292336dbff19c7c/actionview/lib/action_view/helpers/rendering_helper.rb#L35
+    render partial: "local_assigns", locals: { title: "Title", byline: byline, header: tag.h1("Yo") }
+
+    assert_css "h1", text: "Title"
+    assert_css "span", text: "Some guy"
+    assert_match "<h1>Yo</h1>", rendered
+
+    assert_equal "Some guy", byline # We can't mutate the passed in content.
+  end
```

Comment:
@seanpdoyle if I write this test like so:

```ruby
  test "passing local_assigns as content seeds with hash render call" do
    byline = "Some guy"

    render partial: "local_assigns", locals: { title: "Title", byline: byline, header: tag.h1("Yo") } do |partial|
      partial.byline ", who writes stuff"
      partial.header " more"
    end

    assert_css "h1", text: "Title"
    assert_css "span", text: "Some guy, who writes stuff"
    assert_match "<h1>Yo</h1> more", rendered

    assert_equal "Some guy", byline # We can't mutate the passed in content.
  end
```

We end up hitting this branch in `render`, which flatly overrides the `:partial` key.
https://github.com/rails/rails/blob/90a272a1de6f13711943139c4292336dbff19c7c/actionview/lib/action_view/helpers/rendering_helper.rb#L35

Which then triggers a `to_partial_path` error (because `options[:layout]` is nil):

```
E

Error:
RendererTest#test_passing_local_assigns_as_content_seeds_with_hash_render_call:
ArgumentError: 'nil' is not an ActiveModel-compatible object. It must implement :to_partial_path.
    /Users/kasperhansen/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/actionview-7.0.3/lib/action_view/renderer/abstract_renderer.rb:82:in `partial_path'
    /Users/kasperhansen/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/actionview-7.0.3/lib/action_view/renderer/object_renderer.rb:20:in `render_object_derive_partial'
    /Users/kasperhansen/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/actionview-7.0.3/lib/action_view/renderer/renderer.rb:94:in `render_partial_to_object'
    /Users/kasperhansen/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/actionview-7.0.3/lib/action_view/renderer/renderer.rb:53:in `render_partial'
    /Users/kasperhansen/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/actionview-7.0.3/lib/action_view/helpers/rendering_helper.rb:35:in `block in render'
    /Users/kasperhansen/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/actionview-7.0.3/lib/action_view/base.rb:270:in `in_rendering_context'
    /Users/kasperhansen/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/actionview-7.0.3/lib/action_view/helpers/rendering_helper.rb:33:in `render'
    /Users/kasperhansen/code/bullet-train-co/nice_partials/lib/nice_partials/monkey_patch.rb:54:in `render'
    /Users/kasperhansen/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/actionview-7.0.3/lib/action_view/test_case.rb:202:in `render'
    /Users/kasperhansen/.rbenv/versions/3.0.2/lib/ruby/gems/3.0.0/gems/actionview-7.0.3/lib/action_view/test_case.rb:128:in `render'
    /Users/kasperhansen/code/bullet-train-co/nice_partials/test/renderer_test.rb:144:in `block in <class:RendererTest>'


bin/test renderer_test.rb:141
```

If I pass `layout: "local_assigns"` to the `render` call it works as expected.

I have no clue what the branch in `render` is for and I have no idea why we can't pass a block in this test. @seanpdoyle have you seen this before, or would you happen to know what it's guarding for?